### PR TITLE
libucl: add libucl/0.8.1 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libucl/0.8.1**

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/libucl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from conans import CMake, ConanFile, tools
+import functools
+
+required_conan_version = ">=1.33.0"
+
+
+class LibuclConan(ConanFile):
+    name = "libucl"
+    description = "Universal configuration library parser "
```
Comment (theirix) [inline recipes/libucl/all/conanfile.py:9]: trailing space

File: recipes/libucl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/libusb/libusb"
+    url = "https://github.com/conan-io/conan-center-index"
+    topics = ("conan", "libusb", "usb", "device")
```
Comment (theirix) [inline recipes/libucl/all/conanfile.py:13]: conan is not a proper topic, checked by hook KB-H064

**Review by theirix — CHANGES_REQUESTED** [review]

File: recipes/libucl/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/libucl/all/conanfile.py:13]: The topics are also completely wrong.
I used the libusb recipe as a template.
```suggestion
    topics = ("universal", "configuration", "language", "parser", "ucl")
```

File: recipes/libucl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import functools
+
+required_conan_version = ">=1.33.0"
+
+
+class LibuclConan(ConanFile):
+    name = "libucl"
+    description = "Universal configuration library parser "
+    license = "BSD-2-Clause"
+    homepage = "https://github.com/libusb/libusb"
```
Comment (madebr) [inline recipes/libucl/all/conanfile.py:11]:
```suggestion
    homepage = "https://github.com/vstakhov/libucl"
```

File: recipes/libucl/all/conanfile.py (hunk 1 above)
Comment (madebr) [reply recipes/libucl/all/conanfile.py:9]: Thanks for noticing.
```suggestion
    description = "Universal configuration library parser"
```

**Review by theirix — APPROVED** [review]

File: recipes/libucl/all/patches/0002-0.8.1-cmake-add-install+use-find_package.patch [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
++++ CMakeLists.txt
+@@ -4,13 +4,13 @@
+ SET(LIBUCL_VERSION_MAJOR 0)
+ SET(LIBUCL_VERSION_MINOR 5)
+ SET(LIBUCL_VERSION_PATCH 0)
+-
+-SET(LIBUCL_VERSION
+-        "${LIBUCL_VERSION_MAJOR}.${LIBUCL_VERSION_MINOR}.${LIBUCL_VERSION_PATCH}")
++SET(LIBUCL_VERSION "${LIBUCL_VERSION_MAJOR}.${LIBUCL_VERSION_MINOR}.${LIBUCL_VERSION_PATCH}")
+-
```
Comment (prince-chrismc) [inline recipes/libucl/all/patches/0002-0.8.1-cmake-add-install+use-find_package.patch:13]: This seems a little extra 🤔 why did you decide to include this?

Comment (madebr) [reply recipes/libucl/all/patches/0002-0.8.1-cmake-add-install+use-find_package.patch:13]: I did this to let this chunk  be line neutral: number of input lines == number of output lines.
That makes later chunks much easier.

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**